appveyor.yml

```diff
@@ -49,7 +49,7 @@ build_script:
   # '-Djna.nosys=true' is required to avoid kernel32.dll load failure.
   # See SPARK-28759.
   # Ideally we should check the tests related to Hive in SparkR as well (SPARK-31745).
-  - cmd: mvn -DskipTests -Psparkr -Djna.nosys=true package
+  - cmd: mvn -DskipTests -Phadoop-2.7 -Psparkr -Djna.nosys=true package
```

Comment:
Can we upgrade @steveloughran 's Hadoop 3.0 instead?
- https://github.com/apache/spark/blob/master/dev/appveyor-install-dependencies.ps1#L98
```
# ========================== Hadoop bin package
# This must match the version at https://github.com/steveloughran/winutils/tree/master/hadoop-2.7.1
$hadoopVer = "2.7.1"
```

He provides Hadoop 3 also.
- https://github.com/steveloughran/winutils/tree/master/hadoop-3.0.0/bin

Reply:
Ah, sure. Let me try!

Reply:
I made a separate PR for that https://github.com/apache/spark/pull/29042. If that passes the tests, we can merge that one and close this.